<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from arc_robot_arm_urdf.xacro       | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="arc_robot_arm_urdf">
  <material name="silver">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <transmission name="Rev1_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev1_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev2_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev2_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev6_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev6_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev7_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev7">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev7_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev5_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev5_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev4_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev4_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev3_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev3_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="control"/>
  </gazebo>
  <gazebo reference="base_link">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
    <gravity>true</gravity>
  </gazebo>
  <gazebo reference="link_1_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link_2_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="gear2_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="gear1_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="gripper_base_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link_4_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link_3_v1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0061067465353758 0.004245220451651192 0.022482908442763117"/>
      <mass value="0.7538944695406217"/>
      <inertia ixx="0.001048" ixy="-3.7e-05" ixz="1.9e-05" iyy="0.001037" iyz="2.7e-05" izz="0.001633"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_1_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.003004788195336593 -0.005729527054302647 0.01733071933541839"/>
      <mass value="0.5341010747394327"/>
      <inertia ixx="0.0004" ixy="-5.2e-05" ixz="1.2e-05" iyy="0.00041" iyz="6.3e-05" izz="0.000429"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.006279 -0.004346 -0.056179"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_1_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.006279 -0.004346 -0.056179"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_1_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_2_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.003917799059977441 0.007344369307962232 0.05913220031519059"/>
      <mass value="0.5015278384318742"/>
      <inertia ixx="0.000855" ixy="-1.9e-05" ixz="-1.5e-05" iyy="0.000862" iyz="-1.6e-05" izz="7.2e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.014548 0.007395 -0.096499"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_2_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.014548 0.007395 -0.096499"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_2_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="gear2_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.010701097239873197 -0.010248134925377461 0.0025949055535464116"/>
      <mass value="0.027479876879980724"/>
      <inertia ixx="5e-06" ixy="-4e-06" ixz="0.0" iyy="7e-06" iyz="0.0" izz="1.1e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.124808 0.111859 -0.239413"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gear2_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.124808 0.111859 -0.239413"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gear2_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="gear1_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.012476208530257826 -0.007603268645116054 0.0024608496045445682"/>
      <mass value="0.028106280846863212"/>
      <inertia ixx="5e-06" ixy="-4e-06" ixz="0.0" iyy="7e-06" iyz="0.0" izz="1.1e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.142023 0.091215 -0.238377"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gear1_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.142023 0.091215 -0.238377"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gear1_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="gripper_base_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.013700588821693407 -0.013432166948005697 0.005812130546821476"/>
      <mass value="0.10839935246360793"/>
      <inertia ixx="3.7e-05" ixy="-1.2e-05" ixz="5e-06" iyy="2.2e-05" iyz="8e-06" izz="4.6e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.107121 0.074593 -0.223325"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gripper_base_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.107121 0.074593 -0.223325"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/gripper_base_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_4_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.016942437858265463 -0.006965566795622061 0.005991627929883364"/>
      <mass value="0.1295112121194818"/>
      <inertia ixx="2.4e-05" ixy="-8e-06" ixz="1e-06" iyy="3.6e-05" iyz="-0.0" izz="3.1e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.076785 0.067854 -0.217054"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_4_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.076785 0.067854 -0.217054"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_4_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_3_v1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.02543692230594534 -0.03072948509594812 0.0026338069795326935"/>
      <mass value="0.39461490281663475"/>
      <inertia ixx="0.000232" ixy="-0.000188" ixz="1.8e-05" iyy="0.000299" iyz="1.5e-05" izz="0.000427"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.012404 0.004655 -0.216448"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_3_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.012404 0.004655 -0.216448"/>
      <geometry>
        <mesh filename="package://arc_robot_arm_urdf_description/meshes/link_3_v1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="Rev1" type="continuous">
    <origin rpy="0 0 0" xyz="-0.006279 0.004346 0.056179"/>
    <parent link="base_link"/>
    <child link="link_1_v1_1"/>
    <axis xyz="-0.002854 0.005633 0.99998"/>
  </joint>
  <joint name="Rev2" type="continuous">
    <origin rpy="0 0 0" xyz="-0.008269 -0.011741 0.04032"/>
    <parent link="link_1_v1_1"/>
    <child link="link_2_v1_1"/>
    <axis xyz="-0.638789 0.769358 -0.006157"/>
  </joint>
  <joint name="Rev6" type="continuous">
    <origin rpy="0 0 0" xyz="-0.017687 -0.037266 0.016088"/>
    <parent link="gripper_base_v1_1"/>
    <child link="gear2_v1_1"/>
    <axis xyz="0.00405 0.053471 0.998561"/>
  </joint>
  <joint name="Rev7" type="continuous">
    <origin rpy="0 0 0" xyz="-0.034902 -0.016622 0.015052"/>
    <parent link="gripper_base_v1_1"/>
    <child link="gear1_v1_1"/>
    <axis xyz="0.00405 0.053471 0.998561"/>
  </joint>
  <joint name="Rev5" type="continuous">
    <origin rpy="0 0 0" xyz="-0.030336 -0.006739 0.006271"/>
    <parent link="link_4_v1_1"/>
    <child link="gripper_base_v1_1"/>
    <axis xyz="0.639951 -0.767451 0.0385"/>
  </joint>
  <joint name="Rev4" type="continuous">
    <origin rpy="0 0 0" xyz="-0.064381 -0.063199 0.000606"/>
    <parent link="link_3_v1_1"/>
    <child link="link_4_v1_1"/>
    <axis xyz="-0.767993 -0.637134 0.065163"/>
  </joint>
  <joint name="Rev3" type="continuous">
    <origin rpy="0 0 0" xyz="0.002144 0.00274 0.119949"/>
    <parent link="link_2_v1_1"/>
    <child link="link_3_v1_1"/>
    <axis xyz="0.638789 -0.769358 0.006157"/>
  </joint>
</robot>
